pkg/infra/kubernetes/persist_test.go

```diff
@@ -0,0 +1,335 @@
+package kubernetes
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/lang/yaml"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_handlePersistForExecUnit(t *testing.T) {
+	type testResult struct {
+		values []Value
+		file   string
+	}
+	tests := []struct {
+		name           string
+		unit           HelmExecUnit
+		deploymentYaml string
+		podYaml        string
+		deps           []core.Dependency
+		want           testResult
+		wantErr        bool
+	}{
+		{
+			name: "orm dependency and deployment",
+			unit: HelmExecUnit{
+				Name:      "unit",
+				Namespace: "default",
+			},
+			deploymentYaml: `apiVersion: apps/v1
+kind: Deployment
```

Comment:
yeah you always have to use spaces, its a bit annoying, but would be really hard to test all of these cases end to end with integ tests so im just trying to get things in place here. We maybe could write some k8s testing suite to check our compiled output instead of unit tests though